src/client/dfs/dfs_sys.c

```diff
@@ -455,24 +453,26 @@ dfs_sys_umount(dfs_sys_t *dfs_sys)
 			d_hash_rec_decref(dfs_sys->hash, rlink);
 		}
 
-		hash_rc = d_hash_table_destroy(dfs_sys->hash, false);
-		if (hash_rc != 0) {
-			D_DEBUG(DB_TRACE, "failed to destroy hash table: "
-				DF_RC"\n", DP_RC(hash_rc));
+		rc = d_hash_table_destroy(dfs_sys->hash, false);
+		if (rc) {
+			D_DEBUG(DB_TRACE, "failed to destroy hash table: "DF_RC"\n", DP_RC(rc));
+			D_GOTO(out, rc);
 		}
+		dfs_sys->hash = NULL;
```

Comment:
@mchaarawi @ashleypittman How do you feel about setting the struct members to NULL? (here and below for `dfs_sys->dfs`) The idea is that by doing so we can call `dfs_sys_umount` multiple times without failure, since the pointer would be non-NULL, but garbage, on success.

Reply:
makes sense to me